### PR TITLE
PLAT-122887: Add more component views for screenshot test - set 1

### PR DESCRIPTION
**Pull request by stanca-pop-lgp**
Enact-DCO-1.0-Signed-off-by: Stanca Pop stanca.pop@lgepartner.com

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Agate - added more component views for screenshot test (ContextualPopupDecorator, Drawer, FullscreenPopup, Popup, PopupMenu, TooltipDecorator, Header, Heading, Dropdown, Image, Input, Keypad, LabeledIcon, LabeledIconButton, Panels, TabGroup)

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
* Root Cause
=> There were only 16 component views for screenshot test in https://github.com/enactjs/agate/tree/develop/tests/screenshot/apps/components. But there are 45 public ui components.
 Solution -> Added more component views for screenshot test

### Links
[//]: # (Related issues, references)
PLAT-122887

### Comments

### CODE REVIEW

**Review by daniel-stoian-lgp — APPROVED** [review]

File: tests/screenshot/apps/components/Heading.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import React from 'react';
 
 const HeadingTests = [
 	<Heading />,
-	<Heading>tHis is a neW Heading - default</Heading>,
+	<Heading size="title">tHis is a neW Heading - default</Heading>,
```
Comment (adrian-cocoara-lgp) [inline tests/screenshot/apps/components/Heading.js:6]: tHis is a neW Heading -> This is a new Heading

File: tests/screenshot/apps/components/ContextualPoppDecorator.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import ContextualPopupDecorator from '../../../../ContextualPopupDecorator';
+import React from 'react';
+
+const ContextualPopupDecoratorTests = [
+	<ContextualPopupDecorator />,
+	<ContextualPopupDecorator direction="above" />,
+	<ContextualPopupDecorator noArrow />
+];
+
+export default ContextualPopupDecoratorTests;
```
Comment (ybsung) [inline tests/screenshot/apps/components/ContextualPoppDecorator.js:10]: Please check this file name. Not ContextualPoppDecorator but ContextualPopupDecorator.js.

File: tests/screenshot/apps/components/Image.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import Image from '../../../../Image';
+import React from 'react';
+
+const ImageTests = [
+	<Image />,
+	<Image src="http://via.placeholder.com/200x200" />,
+	<Image sizing="fit" src="http://via.placeholder.com/200x200" />
+];
+export default ImageTests;
```
Comment (ybsung) [inline tests/screenshot/apps/components/Image.js:9]: We need one empty line before defining `export default` and the last line in a file.

File: tests/screenshot/apps/components/ContextualPoppDecorator.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import ContextualPopupDecorator from '../../../../ContextualPopupDecorator';
+import React from 'react';
+
+const ContextualPopupDecoratorTests = [
+	<ContextualPopupDecorator />,
+	<ContextualPopupDecorator direction="above" />,
+	<ContextualPopupDecorator noArrow />
```
Comment (ybsung) [inline tests/screenshot/apps/components/ContextualPoppDecorator.js:7]: Decorator should be used as a tag. It is not a component but a HoC. Please check usage from https://github.com/enactjs/agate/blob/develop/samples/qa-a11y/src/views/ContextualPopupDecorator.js#L12 and screenshot result.

File: tests/screenshot/apps/components/Drawer.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import Drawer from '../../../../Drawer';
+import React from 'react';
+
+const DrawerTests = [
+	<Drawer>{[]}</Drawer>,
+	<Drawer noAnimation>{[]}</Drawer>,
+	<Drawer open>{[]}</Drawer>
```
Comment (ybsung) [inline tests/screenshot/apps/components/Drawer.js:7]: I might be better to add `noAnimation` to all Drawer components.
If not, we might easily get fails due to timing like Spinner. Spinner has animation, so it is hard to get successful screenshot test result.

**Review by ybsung — CHANGES_REQUESTED** [review]
image
image

Dropdown, TabGrorup screenshot test fails. Please fix it.

There is nothing in the screenshot of Panels, Popup, PopupMenu, and TooltipDecorator. Please add one test at least to show anything.

File: tests/screenshot/apps/components/TabGroup.js [hunk 6]
```diff
@@ -2,9 +2,20 @@ import TabGroup from '../../../../TabGroup';
 import React from 'react';
 
 const TabGroupTests = [
-	<TabGroup />,
-	<TabGroup orientation="vertical" />,
-	<TabGroup tabPosition="after" />
+	<TabGroup tabPosition="before"
+			  tabs={[
```
Comment (ybsung) [inline tests/screenshot/apps/components/TabGroup.js]: Please check indentation.

File: tests/screenshot/apps/components/Keypad.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import Keypad from '../../../../Keypad';
+import React from 'react';
+
+const KeypadTests = [
+	<Keypad />,
+	<Keypad disabled />
+];
+export default KeypadTests;
```
Comment (ybsung) [inline tests/screenshot/apps/components/Keypad.js:8]: Please add one empty line before this line.

File: tests/screenshot/apps/components/Dropdown.js [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import Dropdown from '../../../../Dropdown';
+import React from 'react';
+
+const DropdownTests = [
+	<Dropdown>{[]}</Dropdown>,
+	<Dropdown direction="up">{[]}</Dropdown>,
+	<Dropdown open>{[]}</Dropdown>
```
Comment (ybsung) [inline tests/screenshot/apps/components/Dropdown.js:7]: Could you add some texts for the Dropdowns to show something in screenshot tests?

**Review by ybsung — APPROVED** [review]
LGTM

File: tests/screenshot/apps/components/Header.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import Header from '../../../../Header';
+import React from 'react';
+
+const HeaderTests = [
```
Comment (jeonghee27) [inline tests/screenshot/apps/components/Header.js:4]: Shouldn't there be a title?

Comment (stanca-pop-lgp) [reply tests/screenshot/apps/components/Header.js:4]: Yes, there should. I fixed it, Thank you.

**Comment by jeonghee27** [comment]
Code is looks good to me. but something is wrong with the branch.
Some unrelated modification is shown in "File changed". It seems that rebase is needed to based on develop. If it's difficult, you'll need to create new branch again based on develop. (In this case, please post a new PR)

**Comment by stanca-pop-lgp** [comment]
@jeonghee27 can you check it now, please?

**Comment by jeonghee27** [comment]
LGTM~

**Pull request merged**